tests/commands/test_update.py

```diff
@@ -32,103 +31,447 @@ def test_update_run(tmp_library: TemporaryLibrary) -> None:
+
+
+def test_update_yaml_cli(
+    tmp_library: TemporaryLibrary,
+    resource_cache: ResourceCache,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    from papis.commands.update import cli
+
     cli_runner = PapisRunner()
 
     filename = os.path.join(resource_cache.cachedir, "update", "russell.yaml")
 
     import papis.utils
+
     with monkeypatch.context() as m:
-        m.setattr(papis.utils, "update_doc_from_data_interactively",
-                  update_doc_from_data_interactively)
+        m.setattr(
+            papis.utils,
+            "update_doc_from_data_interactively",
+            update_doc_from_data_interactively,
+        )
 
-        result = cli_runner.invoke(
-            cli,
-            ["--from", "yaml", filename, "krishnamurti"])
+        result = cli_runner.invoke(cli, ["--from", "yaml", filename, "krishnamurti"])
         assert result.exit_code == 0
 
     import papis.yaml
+
     data = papis.yaml.yaml_to_data(filename)
 
     db = papis.database.get()
-    doc, = db.query_dict({"author": "Russell"})
+    (doc,) = db.query_dict({"author": "Russell"})
 
     assert doc["doi"] == data["doi"]
 
 
-def test_update_bibtex_cli(tmp_library: TemporaryLibrary,
-                           resource_cache: ResourceCache,
-                           monkeypatch: pytest.MonkeyPatch) -> None:
+def test_update_bibtex_cli(
+    tmp_library: TemporaryLibrary,
+    resource_cache: ResourceCache,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
     from papis.commands.update import cli
+
     cli_runner = PapisRunner()
 
     filename = os.path.join(resource_cache.cachedir, "update", "wannier.bib")
 
     import papis.utils
+
     with monkeypatch.context() as m:
-        m.setattr(papis.utils, "update_doc_from_data_interactively",
-                  update_doc_from_data_interactively)
+        m.setattr(
+            papis.utils,
+            "update_doc_from_data_interactively",
+            update_doc_from_data_interactively,
+        )
 
-        result = cli_runner.invoke(
-            cli,
-            ["--from", "bibtex", filename, "krishnamurti"])
+        result = cli_runner.invoke(cli, ["--from", "bibtex", filename, "krishnamurti"])
         assert result.exit_code == 0
 
     import papis.bibtex
-    data, = papis.bibtex.bibtex_to_dict(filename)
+
+    (data,) = papis.bibtex.bibtex_to_dict(filename)
 
     db = papis.database.get()
-    doc, = db.query_dict({"author": "Wannier"})
+    (doc,) = db.query_dict({"author": "Wannier"})
 
     assert doc["doi"] == data["doi"]
-
-
-def test_update_cli_ref(tmp_library: TemporaryLibrary) -> None:
-    from papis.commands.update import cli
-    cli_runner = PapisRunner()
-
-    result = cli_runner.invoke(
-        cli,
-        ["krishnamurti", "--set", "ref", "NewRef"])
-    assert result.exit_code == 0
-    assert not result.output
-
-    db = papis.database.get()
-    doc, = db.query_dict({"author": "krishnamurti"})
-    assert doc["ref"] == "NewRef"
```

Comment:
Why was this removed? Or am I missing something?

Reply:
I removed this as it seemed redundant. Setting a ref is just like setting any string and that's already being tested elsewhere. If you'd like it back, happy to do so :smiley:

Reply:
Hm, yeah, I guess it is redundant now! I added this test in #539 because before `papis update` wouldn't work on refs at all.

Reply:
Ah makes sense! I've made a small change and now explicitly set a `ref` in `test_update_set_general_cli()`. It's probably not a bad thing to have that in there given how important `ref`s are